pkg/server/plugin/upstreamauthority/vault/vault.go

```diff
@@ -58,6 +60,21 @@ type Configuration struct {
 	Namespace string `hcl:"namespace" json:"namespace"`
 }
 
+func NewConfiguration(text string, core *configv1.CoreConfiguration) (config *Configuration, notes []string, err error) {
+	newConfig := new(Configuration)
+	if err := hcl.Decode(newConfig, text); err != nil {
+		notes = append( notes, PluginConfigMalformed )
+		return nil, notes, status.Error(codes.InvalidArgument, notes[0])
+	}
+
+	var unusable bool = false
```

Comment:
I understand that this unusable bool is needed to have appended notes. I believe that current validations in SPIRE returns a single error, even if as a result of continuing with the validation it would return more errors, so it wouldn't be surprising to return a single note.
I'm making this comment just to point that we could simplify the logic just returning a single note, but I don't feel too strongly on it.

Reply:
There are only calls to configure a plugin, there are not current calls to validate a plugin's contents.

The maintainers insisted that for validation, all failures or successful notes be returned in a group, so one doesn't have to validate the plugin config multiple times to just get "the next error".  I agreed with this request, and it was worked into the plugin sdk.  Asking for a change like this is basically asking that the maintainers rescind their prior requests, when they were clearly improvements.  I don't think that's appropriate.

The code above was worked to ensure that the same single note that fails the Configure call is the single returned error under the Configure usage of NewConfiguration.  It also returns multiple notes (some of which might not be errors) in the Validate calls.  This permits one to validate once, getting a list of issues, notes, or errors.

Reply:
@amartinezfayo @azdagron 

In the request to return multiple potential notes (including multiple potential errors) in one call to the gRPC `Validate(...)`, we made the need to return multiple errors, if they could be reported in one pass part of the solution's requirements.